src/TourManager/TourManager.Repository.EfCore.MsSql/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Linq.Expressions;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
+using TourManager.Repository.Abstraction;
+
+namespace TourManager.Repository.EfCore.MsSql.Repositories
+{
+    public class BaseRepository<TEntity> : IRepository<TEntity> where TEntity : class
+    {
+        protected readonly DbContext Context;
+        protected readonly DbSet<TEntity> DbSet;
+
+        public BaseRepository(DbContext context)
+        {
+            Context = context;
+            DbSet = Context.Set<TEntity>();
+        }
+
+        public async Task<TEntity> GetByIdAsync(int id)
+        {
+            return await Context.Set<TEntity>().FindAsync(id);
+        }
+
+        public async Task<IEnumerable<TEntity>> GetAllAsync()
+        {
+            var query = Context.Set<TEntity>().AsNoTracking();
```

Comment:
Why we are using `AsNoTracking() `?

Reply:
No tracking queries are useful when the results are used in a read-only scenario. They're quicker to execute because there's no need to set up the change tracking information. If you don't need to update the entities retrieved from the database, then a no-tracking query should be used.